pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1046,19 +1071,17 @@ func publishΗealthChecksReport(ctx *zedagentContext, iteration int) {
 	mcs, err := edac.MemoryControllers()
 	if err != nil {
 		log.Error(err)
-		return
 	}
 
 	if len(mcs) == 0 {
 		log.Noticef("No ECC memory found in the system")
```

Comment:
On a device without ECC memory it doesn't make sense generating this log every hour.
There is a sync.Once() function which can be used to run something once.

Reply:
I do not think that sync.Once() will help here, I need to differentiate between no ECC memory in the system and the ECC memory report not sent. If I do not have ECC memory in the system, I should return after publishing that empty memory info report. A better approach would be to update EVE-API and add a status in the ECCMemoryReport message.

Reply:
There are two things:
1. How to indicate that there is no ECC memory.
2. We should not do any periodic logging at INFO or above due to expected conditions.

For #2 you can make sure you generate this log only once. sync.Once() was only a suggestion.

#1: Receiving a hardware health message without the ECC memory field set seems like a fine indication that there is no ECC memory. I don't see how an additional status would make any difference (unless you want to report any errors you get when gathering the ECC info??)